package.json

```diff
@@ -24,6 +27,11 @@
     "storybook:clean": "rimraf storybook-static",
     "release": "semantic-release"
   },
+  "husky": {
+    "hooks": {
+      "pre-commit": "yarn ts:type-check && lint-staged"
```

Comment:
I would only call `lint-staged` because we only want to do type checking on staged files, not the whole project.
```
"pre-commit": "lint-staged"
```

Reply:
So I had a go at this, unfortunately, the type checking isn't possible within lint-staged, even when there are unused vars, it still passes. So it seems like TS doesn't support this - https://github.com/okonet/lint-staged/issues/468#issuecomment-418827220 

When I run the type checking outside lint-staged it works well. It takes about 5s on my machine for it to check the whole project, I think this is okay.

Reply:
Should have read that link you originally posted a bit more. This looks like the best solution at the moment 👍